ReactWindows/ChakraBridge/ChakraHost.cpp

```diff
@@ -333,21 +334,50 @@ JsErrorCode ChakraHost::SerializeScript(const wchar_t* szPath, const wchar_t* sz
     ULONG bufferSize = 0L;
     BYTE* buffer = nullptr;
     wchar_t* szScriptBuffer = nullptr;
+    wchar_t* szTmpSerializedPath = nullptr;
     IfFailRet(LoadFileContents(szPath, &szScriptBuffer));
 
+    JsErrorCode status = JsNoError;
     if (!CompareLastWrite(szSerializedPath, szPath))
     {
-        IfFailRet(JsSerializeScript(szScriptBuffer, buffer, &bufferSize));
+        IfFailCleanup(JsSerializeScript(szScriptBuffer, buffer, &bufferSize));
         buffer = new BYTE[bufferSize];
-        IfFailRet(JsSerializeScript(szScriptBuffer, buffer, &bufferSize));
+        IfFailCleanup(JsSerializeScript(szScriptBuffer, buffer, &bufferSize));
+
+        size_t tmpFilePathSize = wcslen(szSerializedPath) + 4; // add string size of ".tmp"
+        szTmpSerializedPath = new wchar_t[tmpFilePathSize + 1];
+        swprintf(szTmpSerializedPath, tmpFilePathSize + 1, L"%ws.tmp", szSerializedPath);
 
         FILE* file;
-        _wfopen_s(&file, szSerializedPath, L"wb");
-        fwrite(buffer, sizeof(BYTE), bufferSize, file);
+        errno_t err;
+        err = _wfopen_s(&file, szTmpSerializedPath, L"wb");
```

Comment:
combine declaration and assignment. also, any reason to avoid use of `auto`?

Reply:
Will change. No reason, perhaps I'm just not very accustomed to it (I'm an old timer)